src/executor.ts

```diff
@@ -1,18 +1,23 @@
 import * as exec from '@actions/exec';
 import * as io from '@actions/io';
 import * as tc from '@actions/tool-cache';
+import * as versionParser from './version-parser'
 
-export async function instrument(agentVersion:string, scopeGradlePluginVersion:string): Promise<void> {
+const scopeAgentMetadataURL = "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/maven-metadata.xml";
+const scopeGradlePluginMetadataURL = "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-gradle-plugin/maven-metadata.xml";
+const scopeGradleInstrMetadataURL = "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-instrumentation-for-gradle/maven-metadata.xml";
+
+export async function instrument(allowBeta:boolean): Promise<void> {
     const workdir = process.cwd();
-    const scopeAgentPath = await tc.downloadTool("https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/"+agentVersion+"/scope-agent-"+agentVersion+".jar");
-    if(!scopeAgentPath.endsWith(".jar")) {
-        await io.mv(scopeAgentPath, scopeAgentPath+".jar");
-    }
 
-    const gradleInstrumentatorPath = await tc.downloadTool("https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-instrumentation-for-gradle/0.1.0/scope-instrumentation-for-gradle-0.1.0.jar");
+    const agentVersion = versionParser.getVersionToUse(scopeAgentMetadataURL, allowBeta);
+    const pluginVersion = versionParser.getVersionToUse(scopeGradlePluginMetadataURL, false);
+    const instrVersion = versionParser.getVersionToUse(scopeGradleInstrMetadataURL, false);
```

Comment:
these return promises but you are not awaiting for them, so I don't think this is going to work:
this should work:
```javascript
const [agentVersion, pluginVersion, instrVersion] = await Promise.all([getVersionToUse(scopeAgentMetadataURL, allowBeta), getVersionToUse(scopeGradlePluginMetadataURL, false), getVersionToUse(scopeGradleInstrMetadataURL, false)])
```